heir_py/mlir_emitter.py

```diff
@@ -0,0 +1,199 @@
+"""Emitter from numba IR (SSA) to MLIR."""
+
+import operator
+import textwrap
+
+from numba.core import ir
+
+
+class TextualMlirEmitter:
+    def __init__(self, ssa_ir):
+        self.ssa_ir = ssa_ir
+        self.temp_var_id = 0
+        self.numba_names_to_ssa_var_names = {}
+        self.globals_map = {}
+
+    def emit(self):
+        func_name = self.ssa_ir.func_id.func_name
+        # probably should use unique name...
```

Comment:
might be something i don't know about numba but aren't all func names supposed to be unique in a module?

Reply:
it's just a python thing, that this func_name is only the inner-most name, not including the module import path to the function. There is a qualified name, but I haven't looked into using it, or if it is even necessary.